src/ILCompiler.ReadyToRun/src/Compiler/ReadyToRunCodegenCompilation.cs

```diff
@@ -62,15 +62,22 @@ private static IEnumerable<ICompilationRootProvider> GetCompilationRoots(IEnumer
 
         protected override void CompileInternal(string outputFile, ObjectDumper dumper)
         {
-            using (FileStream inputFile = File.OpenRead(_inputFilePath))
+            foreach (var inputFilePath in _inputFilePaths)
             {
-                PEReader inputPeReader = new PEReader(inputFile);
+                using (FileStream inputFile = File.OpenRead(inputFilePath))
+                {
+                    PEReader inputPeReader = new PEReader(inputFile);
 
-                _dependencyGraph.ComputeMarkedNodes();
-                var nodes = _dependencyGraph.MarkedNodeList;
+                    _dependencyGraph.ComputeMarkedNodes();
+                    var nodes = _dependencyGraph.MarkedNodeList;
 
-                NodeFactory.SetMarkingComplete();
-                ReadyToRunObjectWriter.EmitObject(inputPeReader, outputFile, nodes, NodeFactory);
+                    NodeFactory.SetMarkingComplete();
+                    if (_inputFilePaths.Count > 1)
+                    {
+                        outputFile = Path.GetDirectoryName(outputFile) + "\\" + Path.GetFileNameWithoutExtension(inputFilePath) + ".ni" + Path.GetExtension(inputFilePath);
```

Comment:
We should not be mangling the output filename explicitly specified on the command line.